cmd/oras/internal/display/metadata/model/pull.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package model
+
+import (
+	"path/filepath"
+	"sync"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/content/file"
+)
+
+// File records metadata of a pulled file.
+type File struct {
+	// Path is the absolute path of the pulled file.
+	Path string
+	Descriptor
+}
+
+// newFile creates a new file metadata.
+func newFile(name string, outputDir string, desc ocispec.Descriptor, descPath string) File {
+	path := name
+	if !filepath.IsAbs(name) {
+		// ignore error since it's successfully written to file store
+		path, _ = filepath.Abs(filepath.Join(outputDir, name))
+	}
+	if desc.Annotations[file.AnnotationUnpack] == "true" {
+		path += string(filepath.Separator)
```

Comment:
What if `name` is in Unix style but `Separator` is in Windows style?

Reply:
According to my test result, unix `Separator` in the `name` has already be converted to Windows' automatically before appending.